cmd/editorconfig-checker/main.go

```diff
@@ -2,56 +2,69 @@
 package main
 
 import (
+	"errors"
 	"flag"
 	"fmt"
+	"io/fs"
 	"os"
 
 	"github.com/editorconfig-checker/editorconfig-checker/v3/pkg/config"
 	"github.com/editorconfig-checker/editorconfig-checker/v3/pkg/error"
 	"github.com/editorconfig-checker/editorconfig-checker/v3/pkg/files"
-	"github.com/editorconfig-checker/editorconfig-checker/v3/pkg/logger"
 	"github.com/editorconfig-checker/editorconfig-checker/v3/pkg/outputformat"
 	"github.com/editorconfig-checker/editorconfig-checker/v3/pkg/utils"
 	"github.com/editorconfig-checker/editorconfig-checker/v3/pkg/validation"
 )
 
-// version is used vor the help
+// version is used for the help and to verify against the version stored in the config file
 // version is dynamically set at compiletime
-var version string
+var version string = "development"
 
 // defaultConfigFileNames determines the file names where the config is located
 var defaultConfigFileNames = []string{".editorconfig-checker.json", ".ecrc"}
 
 // currentConfig is the config used in this run
 var currentConfig *config.Config
 
-// Init function, runs on start automagically
-func init() {
-	var configFilePath string
-	var tmpExclude string
-	var c config.Config
-	var init bool
+// exitStub is there to be replaced while running the tests
+var exitStub = os.Exit
```

Comment:
I won't call it stub here. What you do in the tests is a replacing it with a stub.

The calls made in main func with "exitStub" are very strange to me

Here it's the name that will be called in the code.

I could puld use something like this


```suggestion
var exiter = os.Exit
```

```suggestion
var exitFunc = os.Exit
```

Reply:
You're right, technically speaking this would make this function in main.go `exitProxy`.
I'm not really a fan of `exiter` since it's too close to the stem "exciting", and just appending `Func` is also not my cup of tea. I'd hope you can accept `exitProxy`.